sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -1005,4 +1047,520 @@ export class DataLakeFileClient extends DataLakePathClient {
       span.end();
     }
   }
+
+
+  // high level functions
+
+  /**
+   * ONLY AVAILABLE IN NODE.JS RUNTIME.
+   *
+   * Uploads a local file to a Data Lake file.
+   *
+   * @param {string} filePath Full path of the local file
+   * @param {FileParallelUploadOptions} [options]
+   * @returns {(Promise<FileUploadResponse>)}
+   * @memberof DataLakeFileClient
+   */
+  public async uploadFile(
+    filePath: string,
+    options: FileParallelUploadOptions = {}
+  ): Promise<FileUploadResponse> {
+    const { span, spanOptions } = createSpan("DataLakeFileClient-uploadFile", options.tracingOptions);
+    try {
+      const size = (await fsStat(filePath)).size;
+
+      if (size > FILE_MAX_SIZE_BYTES) {
+        throw new RangeError(`size must be < ${FILE_MAX_SIZE_BYTES}.`);
+      }
+
+      // Create the file.
+      await this.create({
```

Comment:
what if file already existed?

Reply:
upload support both creating new file and overwrite an existing file. Users can control this with the `options.condition`.

Reply:
which condition? might better add examples later to cover this common scenario.

---
In reply to: [388190759](https://github.com/Azure/azure-sdk-for-js/pull/7659#discussion_r388190759) [](ancestors = 388190759)

Reply:
Added test cases to cover overwrite and upload with if-not-exist checking scenarios.